tokendito/okta.py

```diff
@@ -559,9 +559,22 @@ def authorize_request(oauth2_config, oauth2_session_data):
     return authorize_code
 
 
-def generate_oauth2_session_data(url):
+def get_nonce(url):
+    """Get nonce from the org server."""
+    userhome_url = f"{url}/app/UserHome"
+    payload = {"session_hint": "AUTHENTICATED", "iss": urllib.parse.quote(userhome_url, safe="")}
+    response = HTTP_client.get(url, params=payload)
+    # '<script nonce="ABCXXXXXXXXXXXXXXXXYZ" type="text/javascript">'
+    nonce = re.search(r"<script nonce=\"(.*)\"\ ", response.text)
+    if nonce:
+        return nonce.group(1)
+    else:
+        return None
```

Comment:
How about this instead? It does the same, but stylistically similar to the way we get values from `response.text` in other places:

```
  @@ -565,11 +565,15 @@ def get_nonce(url):
       payload = {"session_hint": "AUTHENTICATED", "iss": urllib.parse.quote(userhome_url, safe="")}
       response = HTTP_client.get(url, params=payload)
       # '<script nonce="ABCXXXXXXXXXXXXXXXXYZ" type="text/javascript">'
  -    nonce = re.search(r"<script nonce=\"(.*)\"\ ", response.text)
  -    if nonce:
  -        return nonce.group(1)
  -    else:
  -        return None
  +    nonce = None
  +    pattern = re.compile(r'script nonce="(?P<nonce>.*?)" ', re.MULTILINE)
  +
  +    match = pattern.search(response.text)
  +    if match:
  +        logger.debug(f"Found nonce: {match.group('nonce')}")
  +        nonce = match.group("nonce")
  +
  +    return nonce
```

Reply:
Also, the regex should be non-greedy: `"(.*?)"`, your code is picking up more than the nonce you are looking for.